storage/badger/receipts.go

```diff
@@ -99,9 +91,17 @@ func (r *ExecutionReceipts) ByID(receiptID flow.Identifier) (*flow.ExecutionRece
 	return r.byID(receiptID)(tx)
 }
 
-func (r *ExecutionReceipts) Index(blockID, receiptID flow.Identifier) error {
-	return operation.RetryOnConflict(r.db.Update, func(tx *badger.Txn) error {
+func (r *ExecutionReceipts) ByBlockID(blockID flow.Identifier) ([]*flow.ExecutionReceipt, error) {
+	tx := r.db.NewTransaction(false)
+	defer tx.Discard()
+	return r.byBlockId(tx, blockID)
+}
 
+func NewExecutionReceipts(collector module.CacheMetrics, db *badger.DB, results *ExecutionResults) *ExecutionReceipts {
+	r := newExecutionReceipts(collector, db, results)
+	r.index = func(tx *badger.Txn, receipt *flow.ExecutionReceipt) error {
+		blockID := receipt.ExecutionResult.BlockID
+		receiptID := receipt.ID()
```

Comment:
Still failing when we are trying to index different result. For fork recovery it probably makes sense to update entry if it's already there. Should I do this?

Reply:
1. I've just took what there was already implemented.
2. Completely makes sense. Currently `ExecutionReceipts` and `AllExecutionReceipts` uses different indexes so you can handle it easily.